var/spack/repos/builtin/packages/python/sitecustomize.py

```diff
@@ -0,0 +1,38 @@
+"""
+Hack a bunch of directories onto the front of sys.path
+"""
+
+import os
+import site
+
+## Need to find the `.spack-rpaths`file.
+## But, sys.argv hasn't been set up yet.
+# Here's one way, from :
+# https://bugs.python.org/issue2972
+# WTD on things w/out `/proc/self/cmdline` (e.g. OS X)?
+cmdline = open("/proc/self/cmdline").read()
+cmd = cmdline.split('\x00')[1]
```

Comment:
not the best variable name...

Reply:
That's an understatement.... :scream:

Reply:
**** This is Unix-Only!!!  Surely there must be a Pythonic way to do it.  Why are you reading `/proc/self/cmdline` instead of just looking at `sys.argv`?

Reply:
`sys.argv` hasn't been set up yet when `sitecustomize.py` runs.  See the notes in the PR intro, the comment a couple of lines above and the URL that links to a bug that discusses it.

Reply:
`sys.argv` hasn't been set yet when `sitecustomize.py` runs.  It's explained a couple of lines above where you commented and there's a discussion in the PR (it's the first and longest of the "sticky bits".  The comments in the code point to Python community discussions.  What do you think of the alternate solution in the comments

What do you think about the alternative solution (commented out, starting at line 16)?

Reply:
If you had the Spack `python` package generate this during it's installation, could it write out the location of `.spack-rpaths` then (as a static path but correct for the Spack installation).